### DISCUSSION TITLE
electrical torque

### DISCUSSION THREAD

**Discussion by faraz4714** [Eomys/pyleecan#619] [Q&A]
dear all,
 I hope you are well.
I have a problem with the theory of electrical torque which is calculated in two parts of tutorials: if you look at the last graph of [How to set the operating point](https://www.pyleecan.org/04_tuto_Operating_point.html), you see the graph of electrical torque versus current angle and the other one is last graph of [How to compute currents, voltage and torque using the Electrical Module](https://www.pyleecan.org/05_tuto_Elec.html); they are different a lot. I know that one of them was calculated by the magnetic model and the other was calculated by the electrical model. 
But I want to know why are they different and which one is the torque value that we get from the motor.

I will appreciate it if someone answers me

Best regards,
Faraz Mirali

**Comment by BonneelP** [comment]
Hello,

Indeed the answer is that the first one is computed with the Magnetic model (with FEMM) and the other by the Electrical model. Both plot are compared to the same reference and the Magnetic model returns the correct values. At one point the Electrical module did return the correct torque but for some reason it changed and I didn't had the time to investigate what changed (we don't use the electrical module very often on our side). I think that there is a test on the Electrical module that returns better result. 

Best Regards,
Pierre

**Comment by faraz4714** [comment]
Hello,
Thank you for your help.
I have another question if possible for you please give me the answer.
I think we are able to compute Back emf in the pyleecan However I saw two classes (OutElec and EEC_PMSM) but I can't understand how I can compute these values.
could you please help me and sharing code if you have a time

Best regards,
faraz Mirali

**Comment by BonneelP** [comment]
Hello,

OutMag has a method comp_emf that uses Phi_wind computed by MagFEMM.

Best regards,
Pierre

**Comment by faraz4714** [comment]
Hello,
I hope you give me an answer without any additional effort.
I want to know in the pyleecan it is possible to calculate THD(Total harmonic distortion) for BEMF?
Best regards,
Faraz Mirali

**Reply by BonneelP** [reply]
Hello,

It is not available yet but it should be hard to add. I would recommended either a new method on OutMag (comp_emf_THD) or a new parameter for comp_emf(is_THD=False).

Best regards,
Pierre
